packages/core/src/codewhisperer/util/zipUtil.ts

```diff
@@ -76,12 +99,48 @@ export class ZipUtil {
         zip.addFile(path.join(projectName, relativePath), Buffer.from(content, 'utf-8'))
 
         this._pickedSourceFiles.add(relativePath)
-        this._totalSize += statSync(uri.fsPath).size
+        this._totalSize += (await fsCommon.stat(uri.fsPath)).size
         this._totalLines += content.split(ZipConstants.newlineRegex).length
 
-        if (this.reachSizeLimit(this._totalSize)) {
+        if (this.reachSizeLimit(this._totalSize, CodeWhispererConstants.SecurityScanType.File)) {
             getLogger().error(`Payload size limit reached.`)
-            throw new Error('Payload size limit reached.')
+            throw new ToolkitError('Payload size limit reached.')
+        }
+
+        const zipFilePath = this.getZipDirPath() + CodeWhispererConstants.codeScanZipExt
+        zip.writeZip(zipFilePath)
+        return zipFilePath
+    }
+
+    protected async zipProject(uri: vscode.Uri) {
+        const zip = new admZip()
+
+        const projectName = this.getProjectName(uri)
+        const projectPath = this.getProjectPath(uri)
+        const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+        if (!workspaceFolder) {
+            throw Error('No workspace folder found')
+        }
+
+        const files = await collectFiles([projectPath], [workspaceFolder])
+        for (const file of files) {
+            const fileSize = (await fsCommon.stat(file.fileUri.fsPath)).size
+            if (this.isJavaClassFile(file.fileUri)) {
+                this._pickedBuildFiles.add(file.fileUri.fsPath)
+                this._totalBuildSize += fileSize
+            } else {
+                if (
+                    this.reachSizeLimit(this._totalSize, CodeWhispererConstants.SecurityScanType.Project) ||
+                    this.willReachSizeLimit(this._totalSize, fileSize)
+                ) {
+                    getLogger().error(`Payload size limit reached`)
+                    throw new ToolkitError('Payload size limit reached.')
```

Comment:
NB: errors are automatically logged
```suggestion
                    throw new ToolkitError('Payload size limit reached.')
```